source/atomic_ops.cpp

```diff
@@ -63,7 +62,56 @@ bool atomic_cas<uint32_t>(uint32_t *ptr, uint32_t *expectedCurrentValue, uint32_
 
     return !__STREXW(desiredValue, ptr);
 }
-#endif /* #if (__CORTEX_M >= 0x03) */
+
+uint32_t atomic_incr(uint32_t *valuePtr, uint32_t delta)
+{
+    uint32_t newValue;
+    do {
+        newValue = __LDREXW(valuePtr) + delta;
```

Comment:
Broadly speaking, when the compiler chooses which function to match, templates are second-class citizens, but overloads are first-class citizens. Because of that, there will be less trouble with casting variables to overloads instead of instantiating more template versions.